solidity/contracts/Gravity.sol

```diff
@@ -246,14 +268,14 @@ contract Gravity is ReentrancyGuard {
 	// generated from the new valset.
 	// Anyone can call this function, but they must supply valid signatures of state_powerThreshold of the current valset over
 	// the new valset.
-function updateValset(
+	function updateValset(
 		// The new version of the validator set
 		ValsetArgs calldata _newValset,
 		// The current validators that approve the change
 		ValsetArgs calldata _currentValset,
 		// These are arrays of the parts of the current validator's signatures
 		ValSignature[] calldata _sigs
-	) public virtual {
+	) external whenNotPaused checkWhiteList {
```

Comment:
Think adding `nonReentrant` check here is better.

Reply:
added